references/cli/workflow.go

```diff
@@ -246,6 +249,91 @@ func NewWorkflowDebugCommand(c common.Args, ioStream cmdutil.IOStreams, wargs *W
 	return cmd
 }
 
+// NewWorkflowListCommand create workflow list command
+func NewWorkflowListCommand(c common.Args, ioStream cmdutil.IOStreams, wargs *WorkflowArgs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List running workflows",
+		Long:    "List running workflows",
+		Example: "vela workflow list",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cli, err := c.GetClient()
+			if err != nil {
+				return err
+			}
+			namespace, err := GetFlagNamespaceOrEnv(cmd, c)
+			if err != nil {
+				return err
+			}
+			if AllNamespace {
+				namespace = ""
+			}
+			ctx := context.Background()
+			return printWorkflowList(ctx, cli, namespace, ioStream)
+		},
+	}
+	cmd.Flags().BoolVarP(&AllNamespace, "all-namespaces", "A", false, "If true, check the specified action in all namespaces.")
+	addNamespaceAndEnvArg(cmd)
+	return cmd
+}
+
+func printWorkflowList(ctx context.Context, c client.Reader, namespace string, ioStream cmdutil.IOStreams) error {
+	table, err := buildWorkflowListTable(ctx, c, namespace)
+	if err != nil {
+		return err
+	}
+	ioStream.Info(table.String())
+	return nil
+}
+
+func buildWorkflowListTable(ctx context.Context, c client.Reader, namespace string) (*uitable.Table, error) {
+	table := newUITable()
+	header := []interface{}{"NAME", "TYPE", "PHASE", "START-TIME", "END-TIME"}
+	if AllNamespace {
+		header = append([]interface{}{"NAMESPACE"}, header...)
+	}
+	table.AddRow(header...)
+	applist := v1beta1.ApplicationList{}
```

Comment:
Does this mean that this command will list Application as well as WorkflowRun?
Since Application can be listed by `vela list`, I'm not sure why we still need to list Applications via `vela workflow list`.

Reply:
That's true, I was attempting one place to look at all workflows be it application or workflowrun. As a kubevela developer when i am debugging errors across workflows, I would have to know if its an application or a workflowrun so included the applications as well so as to know status across all workflows in the cluster for example.

Reply:
That's reasonable. Can we add a line above the Applications or Workflows to distinguish those two types to make it more clear?

Reply:
Oh I see there is a field called type to distinguish those two types.